File: redo/darwin/radiobuttons.m
// 11 june 2015
#include "uipriv_darwin.h"

struct radiobuttons {
	uiRadioButtons r;
	NSTextField *dummy;
};

uiDefineControlType(uiRadioButtons, uiTypeRadioButtons, struct radiobuttons)

static uintptr_t radiobuttonsHandle(uiControl *c)
{
	struct radiobuttons *r = (struct radiobuttons *) c;

	return (uintptr_t) (r->dummy);
}

static void radiobuttonsAppend(uiRadioButtons *rr, const char *text)
{
	struct radiobuttons *r = (struct radiobuttons *) rr;

	PUT_CODE_HERE;
	uiControlQueueResize(uiControl(r));
}

uiRadioButtons *uiNewRadioButtons(void)
{
	struct radiobuttons *r;

	r = (struct radiobuttons *) uiNewControl(uiTypeRadioButtons());

	r->dummy = [[NSTextField alloc] initWithFrame:NSZeroRect];
	[r->dummy setStringValue:@"TODO uiRadioButtons not implemented"];
	uiDarwinMakeSingleViewControl(uiControl(r), r->dummy, YES);

	uiControl(r)->Handle = radiobuttonsHandle;

	uiRadioButtons(r)->Append = radiobuttonsAppend;

	return uiRadioButtons(r);
}

File: haiku/radiobuttons.cpp
// 18 november 2015
#include "uipriv_haiku.hpp"

struct uiRadioButtons {
	uiHaikuControl c;
	BStringView *dummy;
};

uiHaikuDefineControl(
	uiRadioButtons,						// type name
	uiRadioButtonsType,						// type function
	dummy								// handle
)

void uiRadioButtonsAppend(uiRadioButtons *r, const char *text)
{
	// TODO
}

uiRadioButtons *uiNewRadioButtons(void)
{
	uiRadioButtons *r;

	r = (uiRadioButtons *) uiNewControl(uiRadioButtonsType());

	r->dummy = new BStringView(NULL, "TODO uiRadioButtons not implemented");

	uiHaikuFinishNewControl(r, uiRadioButtons);

	return r;
}
